controller/models/base_model.go

```diff
@@ -44,6 +44,11 @@ type EntityRetriever interface {
 	BasePreparedListAssociated(id string, typeLoader EntityRetriever, query ast.Query) (*EntityListResult, error)
 
 	GetStore() boltz.CrudStore
+
+	// GetEntityTypeId returns a unique id for the entity type. Some entities may share a storage type, such
+	//                 as fabric and edge services, and fabric and edge routers. However, they should have
+	//                 distinct entity type ids, so we can figure out which controller to route commands to
```

Comment:
This extra loading space isn't something I've seen in the godoc standards.

https://go.dev/blog/godoc